### PR TITLE
database: updated doc.go

### PR DESCRIPTION
**Pull request by ehnuje**
## Proposed changes

- This added detailed information of storage/database package to storage/database/doc.go 

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [x] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [x] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: storage/database/doc.go [hunk 1]
```diff
@@ -14,5 +14,28 @@
 // You should have received a copy of the GNU Lesser General Public License
 // along with the klaytn library. If not, see <http://www.gnu.org/licenses/>.
 
-// Package database implements various types of databased used in Klaytn.
+/*
+Package database implements various types of databases used in Klaytn.
+This package is used to read/write data from/to the persistent layer.
+
+DBManager is the interface used by the consumers of database package.
```
Comment (jimni1222) [inline storage/database/doc.go]: How about add one header for those explaination?

File: storage/database/doc.go [hunk 2]
```diff
@@ -14,5 +14,28 @@
...
+
+  - badger_database.go       : implementation of badgerDB, which wraps github.com/dgraph-io/badger
+  - cache_manager.go         : implementation of cacheManager, which manages cache layer over persistent layer
+  - db_manager.go            : contains DBManager and databaseManager
+  - interface.go             : interfaces used outside database package
+  - leveldb_database.go      : implementation of levelDB, which wraps github.com/syndtr/goleveldb
+  - memory_database.go       : implementation of MemDB, which wraps go native map structure
+  - metrics.go               : metrics used in database package, mostly related to cacheManager
+  - partitioned_database.go  : implementation of partitionedDB, which wraps a list of Database interface
+  - schema.go                :
```
Comment (jeonghonah) [inline storage/database/doc.go]: Does not `schema.go` need any description?

File: storage/database/doc.go (hunk 1 above)
Comment (ehnuje) [reply storage/database/doc.go]: Good point =) I will add one.

File: storage/database/doc.go (hunk 2 above)
Comment (ehnuje) [reply storage/database/doc.go]: My bad =( I will update this.

File: storage/database/doc.go [hunk 3]
```diff
@@ -14,5 +14,30 @@
...
+  - badger_database.go       : implementation of badgerDB, which wraps github.com/dgraph-io/badger
+  - cache_manager.go         : implementation of cacheManager, which manages cache layer over persistent layer
+  - db_manager.go            : contains DBManager and databaseManager
+  - interface.go             : interfaces used outside database package
+  - leveldb_database.go      : implementation of levelDB, which wraps github.com/syndtr/goleveldb
+  - memory_database.go       : implementation of MemDB, which wraps go native map structure
+  - metrics.go               : metrics used in database package, mostly related to cacheManager
+  - partitioned_database.go  : implementation of partitionedDB, which wraps a list of Database interface
+  - schema.go                : prefixes and suffixes for database keys and database key generating functions
+
```
Comment (nogang) [inline storage/database/doc.go]: There seems to be no need for spaces here.

File: storage/database/doc.go [hunk 4]
```diff
@@ -14,5 +14,30 @@
...
+*/
+
```
Comment (nogang) [inline storage/database/doc.go]: There seems to be no need for spaces here.

**Pull request merged**